src/model/test/CoilUserDefined_GTest.cpp

```diff
@@ -0,0 +1,246 @@
+  ::testing::FLAGS_gtest_death_test_style = "threadsafe";
+
+  ASSERT_EXIT(
+    {
+      Model m;
+      CoilUserDefined coil_ud(m);
+
+      exit(0);
+    },
+    ::testing::ExitedWithCode(0), "");
+}
+
+TEST_F(ModelFixture, CoilUserDefined_constructor) {
+  Model m;
+
+  CoilUserDefined b1(m);
+
+  EXPECT_TRUE(b1.airOutletTemperatureActuator());
+  EXPECT_TRUE(b1.airOutletHumidityRatioActuator());
+  EXPECT_TRUE(b1.airMassFlowRateActuator());
+  EXPECT_TRUE(b1.plantMinimumMassFlowRateActuator());
+  EXPECT_TRUE(b1.plantMaximumMassFlowRateActuator());
+  EXPECT_TRUE(b1.plantDesignVolumeFlowRateActuator());
+  EXPECT_TRUE(b1.plantMassFlowRateActuator());
+  EXPECT_TRUE(b1.plantOutletTemperatureActuator());
+
+  EXPECT_TRUE(b1.airOutletTemperatureActuator().get().actuatedComponent());
+  EXPECT_EQ(b1.handle(), b1.airOutletTemperatureActuator().get().actuatedComponent().get().handle());
+}
+
+TEST_F(ModelFixture, CoilUserDefined_programs) {
+  Model m;
+
+  //PlantLoop plant(m);
+  CoilUserDefined b1(m);
+  EnergyManagementSystemProgramCallingManager mainPCM(m);
+  EnergyManagementSystemProgramCallingManager initPCM(m);
+  EXPECT_TRUE(b1.setOverallModelSimulationProgramCallingManager(mainPCM));
+  EXPECT_EQ(mainPCM, b1.overallModelSimulationProgramCallingManager().get());
+  EXPECT_TRUE(b1.setModelSetupandSizingProgramCallingManager(initPCM));
+  EXPECT_EQ(initPCM, b1.modelSetupandSizingProgramCallingManager().get());
+  b1.resetOverallModelSimulationProgramCallingManager();
+  b1.resetModelSetupandSizingProgramCallingManager();
+  EXPECT_FALSE(b1.overallModelSimulationProgramCallingManager());
+  EXPECT_FALSE(b1.modelSetupandSizingProgramCallingManager());
+}
+
+TEST_F(ModelFixture, CoilUserDefined_addToNode) {
+  Model m;
+
+  CoilUserDefined coil(m);
+
+  model::AirLoopHVAC airLoop(m);
+
+  model::Node supplyOutletNode = airLoop.supplyOutletNode();
+
+  coil.addToNode(supplyOutletNode);
+
+  ASSERT_EQ((unsigned)3, airLoop.supplyComponents().size());
+}
+
+// Add CoilUserDefined to AirLoopHVAC and PlantLoop
+// Test CoilUserDefined::remove and make sure that the loops are intact
+TEST_F(ModelFixture, CoilUserDefined_remove) {
+  Model m;
+  CoilUserDefined coil(m);
+
+  AirLoopHVAC airLoop(m);
+  Node supplyOutletNode = airLoop.supplyOutletNode();
+
+  coil.addToNode(supplyOutletNode);
+  coil.remove();
+  ASSERT_EQ((unsigned)2, airLoop.supplyComponents().size());
+
+  ASSERT_TRUE(m.getConcreteModelObjects<CoilUserDefined>().empty());
+
+  CoilUserDefined coil2(m);
+  coil2.addToNode(supplyOutletNode);
+
+  PlantLoop plant(m);
+  plant.addDemandBranchForComponent(coil2);
+
+  ASSERT_EQ((unsigned)1, m.getConcreteModelObjects<CoilUserDefined>().size());
+  ASSERT_EQ((unsigned)1, plant.demandComponents(CoilUserDefined::iddObjectType()).size());
+
+  coil2.remove();
+  ASSERT_TRUE(m.getConcreteModelObjects<CoilUserDefined>().empty());
+}
+
+// Add ONLY to PlantLoop
+// Test CoilUserDefined::remove and make sure that the plant is intact
```

Comment:
not seeing where you test the plantloop is intact?

Reply:
probably just missing a `ASSERT_TRUE(plant.demandComponents(CoilUserDefined::iddObjectType()).empty());` after the remove